benchmark/src/DriverBench/ParallelGridFSBench.php

```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    /**
+     * GridFS multi-file upload
+     *
+     * @see https://github.com/mongodb/specifications/blob/ddfc8b583d49aaf8c4c19fa01255afb66b36b92e/source/benchmarking/benchmarking.rst#gridfs-multi-file-upload
+     */
+    #[BeforeMethods('beforeUpload')]
+    public function benchUpload(): void
+    {
+        $pids = [];
+        foreach (self::getFileNames() as $file) {
+            $pid = pcntl_fork();
+            if ($pid === 0) {
+                Utils::getDatabase()->selectGridFSBucket()->uploadFromStream(basename($file), fopen($file, 'r'));
+
+                // Exit the child process
+                exit(0);
+            }
+
+            if ($pid === -1) {
+                throw new RuntimeException('Failed to fork');
+            }
+
+            // Keep the forked process id to wait for it later
+            $pids[$pid] = true;
+        }
+
+        // Wait for all child processes to finish
+        while ($pids !== []) {
+            $pid = pcntl_waitpid(-1, $status);
+            unset($pids[$pid]);
+        }
+    }
+
+    public function beforeUpload(): void
+    {
+        foreach (self::getFileNames() as $file) {
+            stream_copy_to_stream(Data::getStream(5 * 1024 * 1024), fopen($file, 'w'));
+        }
+
+        $database = Utils::getDatabase();
+        $database->drop();
+
+        $bucket = $database->selectGridFSBucket();
+        $bucket->uploadFromStream('init', Data::getStream(1));
+
+        Utils::reset();
+    }
+
+    /**
+     * GridFS multi-file download
+     *
+     * @see https://github.com/mongodb/specifications/blob/ddfc8b583d49aaf8c4c19fa01255afb66b36b92e/source/benchmarking/benchmarking.rst#gridfs-multi-file-download
+     */
+    #[BeforeMethods('beforeDownload')]
+    public function benchDownload(): void
+    {
+        $pids = [];
+        foreach (self::getFileNames() as $file) {
+            $pid = pcntl_fork();
+            if ($pid === 0) {
+                $stream = Utils::getDatabase()
+                    ->selectGridFSBucket()
+                    ->openDownloadStreamByName(basename($file));
+                stream_copy_to_stream($stream, fopen($file, 'w'));
+
+                // Exit the child process
+                exit(0);
+            }
+
+            if ($pid === -1) {
+                throw new RuntimeException('Failed to fork');
+            }
+
+            // Keep the forked process id to wait for it later
+            $pids[$pid] = true;
+        }
+
+        // Wait for all child processes to finish
+        while ($pids !== []) {
+            $pid = pcntl_waitpid(-1, $status);
+            unset($pids[$pid]);
+        }
+    }
+
+    public function beforeDownload(): void
+    {
+        // Initialize the GridFS bucket with the files
+        $this->beforeUpload();
+        $this->benchUpload();
+    }
+
+    private static function getFileNames(): array
+    {
+        $tempDir = sys_get_temp_dir() . '/mongodb-php-benchmark';
```

Comment:
The tmp directory is a system setting. It can be changed with `sys_temp_dir` ini setting or with the `TMPDIR` env var.
I don't think anyone will ever need to change it.

Reply:
I was mostly referring to `mongodb-php-benchmark` since it's repeated across tests. No worries.